src/ratings/RatingsClient.js

```diff
@@ -0,0 +1,102 @@
+const puppeteer = require('puppeteer');
+const Promise = require('bluebird');
+const PairedProductList = require('./../PairedProductList');
+/* Class that retrieves the ratings and review info for Amazon products
+   This uses a hack that stands up a bare-bones chrome page for item ratings and review using
+   the item's ASIN.
+*/
+class RatingsClient {
+  /*
+    Instance Attributes:
+      base_url  - base URL to get review and ratings from.
+      delayTime - Amount of time (ms) to delay in between requests. This can be useful when making
+                  hundreds of requests since there is a limit to how many can run asynchronously at once.
+  */
+  constructor() {
+    this.base_url = 'https://www.amazon.com/gp/customer-reviews/widgets/average-customer-review/popover/ref=dpx_acr_pop_?contextId=dpx&asin=';
+    this.delayTime = 1000; // 1s delay.
+  }
+  
+  /*
+    Function that aggregates calls to get all items' information
+  */
+  async getAllItemsRatingsAndReviews(pairedProductList){
+    return this._allProdsReturn(pairedProductList).then((allProds) => {
+      let allProducts = new PairedProductList();
+      allProds.forEach((product) => {
+        allProducts.addPairedProduct(product.amazonProd, product.walmartProd);
+      })
+      return allProducts;
+    });
+  }
+  /*
+    delay function
+  */
+  _timeout(ms) {
+    return new Promise(resolve => setTimeout(resolve, ms));
+  }
+
+  /*
+    Return all products' ratings and reviews
+   */
+  _allProdsReturn(pairedProductList) {
+    return new Promise(async (resolve, reject) => {
+      let promises = [];
+      let index=0;
+      const browser = await puppeteer.launch({headless:false});
```

Comment:
Good point. This took me over a month to find a work around because of the following reasons:
1. I couldn't just do a fetch/get to AMZN because it returned an error saying I couldn't call from a non-browser.
2. I tried using the website you built/AMZN API but they were requiring more business information that we didn't have.
3. I resorted to spinning a browser with a head (headless is set to false) to get this done. I tried to use a headless browser which will not show the chrome app in action but that gave lots of timeout errors. I have found this solution to always work since I arrived at it but if you have other alternative solutions, I am open to  it.

Reply:
I think it's fine the way it is then. There isn't much time until the api key expires anyway.